usama/sprint1/cdk.out/asset.0e9db55350e39158be9550d074e22f47df7cc37733a302a79954c0f2729738d0/web_health_publisher.py

```diff
@@ -0,0 +1,45 @@
+import datetime
+import urllib3
+import urllib.request
+import json
+import constants
+
+
+def health_monitor(event, context):
+    
+    url = constants.URLS_TO_MONITOR
+    
+    availability = get_availability(url)
+    
+    latency = get_latency(url)
+    
+    result = {'Availability': availability, 'Latency': "{}ms".format(latency)}
+    
+    return result
+
+
+
+def get_availability(web_url):
+
+    res = urllib.request.urlopen(urllib.request.Request(
+        url=web_url,
+        headers={'Accept': 'application/json'},
+        method='GET'),
+    timeout=5)
+    
+    availability = res.status
+    
+    return 1 if availability==200 else 0
+    
+def get_latency(web_url):
+    
+    http = urllib3.PoolManager()
+
+    start = datetime.datetime.now()
+    response = http.request('GET', web_url)
+    end = datetime.datetime.now()
+    delta = end - start
+    
+    latency_in_ms = round(delta.microseconds * .000001, 6)
+    return latency_in_ms
```

Comment:
**latency_in_ms** it looks like mili seconds (ms confusing ) but you're measuring in micro_seconds. It's better to change its name.

Reply:
Resolved